src/lavinmq/amqp/client.cr

```diff
@@ -502,16 +503,16 @@ module LavinMQ
       end
 
       private def declare_exchange(frame)
-        if !valid_entity_name(frame.exchange_name)
+        if !NameValidator.valid_entity_name(frame.exchange_name)
           send_precondition_failed(frame, "Exchange name isn't valid")
         elsif frame.exchange_name.empty?
           send_access_refused(frame, "Not allowed to declare the default exchange")
         elsif e = @vhost.exchanges.fetch(frame.exchange_name, nil)
           redeclare_exchange(e, frame)
         elsif frame.passive
           send_not_found(frame, "Exchange '#{frame.exchange_name}' doesn't exists")
-        elsif frame.exchange_name.starts_with? "amq."
-          send_access_refused(frame, "Not allowed to use the amq. prefix")
+        elsif NameValidator.valid_prefix?(frame.exchange_name)
+          send_access_refused(frame, "Not allowed to use that prefix")
```

Comment:
Maybe a little bit confiusing to ask if it's valid, and if it is send an error. Maybe it should be `NameValidator.reserved_prefix?(frame.exchange_name)` instead?